### PR TITLE
build with -DIMGUI_DISABLE_OBSOLETE_KEYIO

### LINKED ISSUE #72102: Crash in imgui (maybe around keyboard layout switching?)
### Describe the bug

I think this might be happening around when I switch keyboard layouts?

```
Assertion failed: ((io.KeysDown[n] == false || IsKeyDown((ImGuiKey)n)) && "Backend needs to either only use io.AddKeyEvent(), either only fill legacy io.KeysDown[] + io.KeyMap[]. Not both!"), function UpdateKeyboardInputs, file imgui.cpp, line 8422.
```

### Attach save file

N/A

### Steps to reproduce

Haven't been able to reliably repro yet.

### Expected behavior

No random crashing in imgui, especially when imgui is not enabled 😅 

### Screenshots

_No response_

### Versions and configuration

- OS: MacOs
    - OS Version: Name: macOS; Version: 14.3; Build: 23D56; 
- Game Version: 0.G-7695-g60757df3a8 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nornagon**
#### Summary
Bugfixes "Occasional crash in imgui when switching keyboard layouts"

#### Purpose of change

Fixes #72102. I didn't narrow down a specific cause, but I wasn't able to repro
with this build flag enabled. It seems like a bug in ImGui's synchronization
between the old input system and the new one, but I'm not totally sure. Anyway,
the input systems can't get desynced if there's only one!

This might break ImTui? I think it still uses the old input system, but I'm not
totally sure.

### CODE REVIEW

**Comment by katemonster33** [comment]
limit this to SDL builds only, and change CMake and MSVC builds too.

File: Makefile [hunk 1]
```diff
@@ -955,6 +955,7 @@ SOURCES += $(THIRD_PARTY_SOURCES)
 
 IMGUI_SOURCES = $(IMGUI_DIR)/imgui.cpp $(IMGUI_DIR)/imgui_demo.cpp $(IMGUI_DIR)/imgui_draw.cpp $(IMGUI_DIR)/imgui_tables.cpp $(IMGUI_DIR)/imgui_widgets.cpp
 
+OTHERS += -DIMGUI_DISABLE_OBSOLETE_KEYIO
 ifeq ($(SDL), 1)
 	IMGUI_SOURCES += $(IMGUI_DIR)/imgui_impl_sdl2.cpp $(IMGUI_DIR)/imgui_impl_sdlrenderer2.cpp
 else
```
Comment (ZhilkinSerg) [inline Makefile:961]:
```suggestion
ifeq ($(SDL), 1)
	OTHERS += -DIMGUI_DISABLE_OBSOLETE_KEYIO
	IMGUI_SOURCES += $(IMGUI_DIR)/imgui_impl_sdl2.cpp $(IMGUI_DIR)/imgui_impl_sdlrenderer2.cpp
else
```

**Comment by nornagon** [comment]
I did cmake as well but i'm not sure where to update the msvc builds.

**Comment by akrieger** [comment]
> I did cmake as well but i'm not sure where to update the msvc builds.

msvc here https://github.com/CleverRaven/Cataclysm-DDA/blob/master/msvc-full-features/Cataclysm-common.props#L58

**Comment by akrieger** [comment]
msvc notiles is broken anyway so the lack of gating around NOTILES is okay.

**Comment by andrei8l** [comment]
This needs to be added to imconfig.h as well, otherwise you hit an assertion failure about mismatched config
https://github.com/CleverRaven/Cataclysm-DDA/blob/4b802fa9f5660e2e64961bd91d5312099931c79a/src/third-party/imgui/imconfig.h#L31

**Comment by nornagon** [comment]
I don't think that's true; if you're getting that assert then something's messed up in your build. Try a clean build?

**Pull request merged**